instrumentation/opentelemetry-instrumentation-redis/tests/test_redis.py

```diff
@@ -226,3 +231,83 @@ def test_no_op_tracer_provider(self):
 
         spans = self.memory_exporter.get_finished_spans()
         self.assertEqual(len(spans), 0)
+
+    def test_attributes_default(self):
+        redis_client = redis.Redis()
+
+        with mock.patch.object(redis_client, "connection"):
+            redis_client.set("key", "value")
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 1)
+
+        span = spans[0]
+        self.assertEqual(
+            span.attributes[SpanAttributes.DB_SYSTEM],
+            DbSystemValues.REDIS.value,
+        )
+        self.assertEqual(
+            span.attributes[SpanAttributes.DB_REDIS_DATABASE_INDEX], 0
+        )
+        self.assertEqual(
+            span.attributes[SpanAttributes.NET_PEER_NAME], "localhost"
+        )
+        self.assertEqual(span.attributes[SpanAttributes.NET_PEER_PORT], 6379)
+        self.assertEqual(
+            span.attributes[SpanAttributes.NET_TRANSPORT],
+            NetTransportValues.IP_TCP.value,
+        )
+
+    def test_attributes_tcp(self):
+        redis_client = redis.Redis.from_url("redis://foo:bar@1.1.1.1:6380/1")
+
+        with mock.patch.object(redis_client, "connection"):
+            redis_client.set("key", "value")
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 1)
+
+        span = spans[0]
+        self.assertEqual(
+            span.attributes[SpanAttributes.DB_SYSTEM],
+            DbSystemValues.REDIS.value,
+        )
+        self.assertEqual(
+            span.attributes[SpanAttributes.DB_REDIS_DATABASE_INDEX], 1
+        )
+        self.assertEqual(
+            span.attributes[SpanAttributes.NET_PEER_NAME], "1.1.1.1"
+        )
+        self.assertEqual(span.attributes[SpanAttributes.NET_PEER_PORT], 6380)
+        self.assertEqual(
+            span.attributes[SpanAttributes.NET_TRANSPORT],
+            NetTransportValues.IP_TCP.value,
+        )
+
+    def test_attributes_unix_socket(self):
+        redis_client = redis.Redis.from_url(
+            "unix://foo@/path/to/socket.sock?db=3&password=bar"
+        )
+
+        with mock.patch.object(redis_client, "connection"):
```

Comment:
What do you think on adding an `assert_called_once_with` on this mock so we can check the connection parameters?

Reply:
Sorry for late answer.

I looked at the code, connection parameters are not passed in this place